demo/src/checkout-pricing.js

```diff
@@ -14,44 +14,72 @@ export function CheckoutPricing () {
     setRecurlyError(null);
     setPlan(plan);
-    setPricing(prev => ({ ...prev, subscriptions: [{ plan, quantity }] }));
+    setPricing(prev => ({ ...prev, subscriptions: [{ plan, quantity: planQuantity }] }));
   };
 
-  function updateQuantity (e) {
-    const quantity = e.target.value;
+  function updatePlanQuantity (e) {
+    const planQuantity = e.target.value;
     setRecurlyError(null);
-    setQuantity(() => quantity);
-    setPricing(prev => ({ ...prev, subscriptions: [{ plan, quantity }] }));
+    setPlanQuantity(planQuantity);
+    setPricing(prev => ({ ...prev, subscriptions: [{ plan, quantity: planQuantity }] }));
   };
 
+  function handleSubmit(e) {
+    e.preventDefault();
+    setPricing({
+      subscriptions: [{ plan, quantity: planQuantity }],
+      adjustments: [{ itemCode, quantity: itemQuantity }],
+    });
+  }
+
   return (
     <div className="DemoSection">
-      <div>
-        <select value={plan} onChange={updatePlan}>
-          <option value="">Select a plan</option>
-          <option value="basic">Basic</option>
-          <option value="advanced">Advanced</option>
-          <option value="error">Error</option>
-        </select>
-        <input
-          type="number"
-          value={quantity}
-          onChange={updateQuantity}
-          placeholder="Quantity"
-          min="0"
-        />
-      </div>
-      <div style={{ marginTop: '15px', visibility: loading ? "hidden" : "" }}>
-        {recurlyError ? <span style={{ color: 'red' }}>{recurlyError.message}</span> : ''}
-        {price && price.now && !recurlyError ? <span>Subtotal: ${price.now.subtotal}</span> : ''}
-      </div>
+      <form onSubmit={handleSubmit}>
+        <div>
+          <select value={plan} onChange={updatePlan}>
+            <option value="">Select a plan</option>
+            <option value="basic">Basic</option>
+            <option value="advanced">Advanced</option>
+            <option value="error">Error</option>
+          </select>
+          <input
+            type="number"
+            value={planQuantity}
+            onChange={updatePlanQuantity}
+            placeholder="Quantity"
+            min="0"
+          />
+          <div style={{ marginTop: '15px' }}>
+            <input
+              type="text"
+              value={itemCode}
+              onChange={e => setItemCode(e.target.value)}
+              placeholder="Item code"
+            />
+            <input
+              type="number"
+              onChange={itemQuantity}
+              onChange={e => setItemQuantity(e.target.value)}
+              placeholder="Quantity"
+              min="0"
+            />
+          </div>
+        </div>
+        <div style={{ marginTop: '15px' }}>
+          {recurlyError ? <span style={{ color: 'red' }}>{recurlyError.message}</span> : ''}
+          {price && price.now && !recurlyError ? <span>Subtotal: ${price.now.subtotal}</span> : ''}
+        </div>
+        <button>Calculate subtotal</button>
```

Comment:
Is there a way to trigger recalculation onChange of any input without having to pepper onChange bindings throughout? I think that would be a more useful UX to demonstrate.

Reply:
We can create a reusable input component for something like this. My intention was to abstract as little as possible, but the demo is definitely getting long and verbose.

Reply:
I also think the demo generally needs some love. I'm thinking a separate PR everything else is merged in to clean up some demo-specific stuff is needed.